File: PabellonWebAPI/BussinessLogicLayer/Services/Stores/StoreService.cs
using BussinessLogicLayer.Reponses;
using BussinessLogicLayer.Request;
using Pabellon.Context.Core.Models;
using Pabellon.Context.Core.Repositories.StoreRepository;
using System.Threading.Tasks;

namespace BussinessLogicLayer.Services.Stores
{
    public class StoreService : IStoreService
    {
        private readonly IStoreRepository _repository;

        public StoreService(IStoreRepository repository)
        {
            _repository = repository;
        }

        public async Task<StoreResponse?> GetStore()
        {
            var store = await _repository.GetStore();
            if (store == null) return null;
            return new StoreResponse
            {
                Name = store.Name,
                PhoneNumber = store.PhoneNumber,
                Address = store.Address,
                Description = store.Description
            };
        }

        public async Task<int> SaveStore(StoreRequest request)
        {
            var store = new Store
            {
                Name = request.Name,
                PhoneNumber = request.PhoneNumber,
                Address = request.Address,
                Description = request.Description
            };
            return await _repository.SaveStore(store);
        }
    }
}

File: PabellonFrontEnd/src/app/services/Entities/store.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { BaseService } from 'src/app/Helpers/BaseService';
import { Store } from 'src/app/models/Store';
import { StoreRequest } from 'src/app/models/Request/StoreRequest';

@Injectable({
  providedIn: 'root'
})
export class StoreService {
  private apiUrl = `${BaseService.baseUrl}/store`;

  constructor(private http: HttpClient) {}

  getStore(): Observable<Store> {
    return this.http.get<Store>(this.apiUrl);
  }

  saveStore(req: StoreRequest): Observable<any> {
    const formData = new FormData();
    formData.append('Name', req.Name);
    formData.append('PhoneNumber', req.PhoneNumber);
    formData.append('Address', req.Address);
    formData.append('Description', req.Description);
    return this.http.post(this.apiUrl, formData);
  }
}
